schemas/iso19139/src/main/plugin/iso19139/process/thumbnail-add.xsl

```diff
@@ -41,55 +44,52 @@
   <xsl:param name="thumbnail_desc" select="''"/>
   <xsl:param name="thumbnail_type" select="''"/>
 
-  <!-- Target element to update. The key is based on the concatenation
-  of URL+Name -->
-  <xsl:param name="updateKey"/>
-
-  <xsl:template match="gmd:identificationInfo/*">
+  <!-- Target element to update.
+    updateKey is used to identify the resource name to be updated - it is for backwards compatibility.  Will not be used if resourceHash is set.
+              The key is based on the concatenation of URL+Name
+    resourceHash is hash value of the object to be removed which will ensure the correct value is removed. It will override the usage of updateKey
+    resourceIdx is the index location of the object to be removed - can be used when duplicate entries exists to ensure the correct one is removed.
+-->
+  <xsl:param name="updateKey" select="''"/>
+  <xsl:param name="resourceHash" select="''"/>
+  <xsl:param name="resourceIdx" select="''"/>
+
+  <xsl:variable name="update_flag">
+    <xsl:value-of select="boolean($updateKey != '' or $resourceHash != '' or $resourceIdx != '')"/>
+  </xsl:variable>
+
+  <!--  Add new gmd:graphicOverview -->
+  <xsl:template match="gmd:identificationInfo/*[$update_flag = false()]">
     <xsl:copy>
-      <xsl:copy-of select="@*"/>
-      <xsl:apply-templates select="gmd:citation"/>
-      <xsl:apply-templates select="gmd:abstract"/>
-      <xsl:apply-templates select="gmd:purpose"/>
-      <xsl:apply-templates select="gmd:credit"/>
-      <xsl:apply-templates select="gmd:status"/>
-      <xsl:apply-templates select="gmd:pointOfContact"/>
-      <xsl:apply-templates select="gmd:resourceMaintenance"/>
-
-      <xsl:if test="$updateKey = ''">
-        <xsl:call-template name="fill"/>
-      </xsl:if>
-
-      <xsl:apply-templates select="gmd:graphicOverview"/>
-
-      <xsl:apply-templates select="gmd:resourceFormat"/>
-      <xsl:apply-templates select="gmd:descriptiveKeywords"/>
-      <xsl:apply-templates select="gmd:resourceSpecificUsage"/>
-      <xsl:apply-templates select="gmd:resourceConstraints"/>
-      <xsl:apply-templates select="gmd:aggregationInfo"/>
-      <xsl:apply-templates select="gmd:spatialRepresentationType"/>
-      <xsl:apply-templates select="gmd:spatialResolution"/>
-      <xsl:apply-templates select="gmd:language"/>
-      <xsl:apply-templates select="gmd:characterSet"/>
-      <xsl:apply-templates select="gmd:topicCategory"/>
-      <xsl:apply-templates select="gmd:environmentDescription"/>
-      <xsl:apply-templates select="gmd:extent"/>
-      <xsl:apply-templates select="gmd:supplementalInformation"/>
-
-      <xsl:apply-templates select="srv:*"/>
-
-      <xsl:apply-templates select="*[namespace-uri()!='http://www.isotc211.org/2005/gmd' and
-                                     namespace-uri()!='http://www.isotc211.org/2005/srv']"/>
+      <xsl:apply-templates select="node()|@*"/>
+      <xsl:call-template name="fill"/>
```

Comment:
Same problems with elements ordering as indicated in previous items.

Please check the XSD documentation (http://www.datypic.com/sc/niem21/e-gmd_MD_DataIdentification.html), the order of the elements is mandatory in the XML structure.

Reply:
@josegar74 I helped fix the element ordering - they should be correct now.